### PR TITLE
fix(call): error-handling-for-empty-track-while-answer-dial

### PR DESCRIPTION
**Pull request by Shreyas281299**
# COMPLETES #[SPARK-471758](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-471758)
## This pull request addresses

Developers could pass empty localAudioStream to dial and answer API which led to roapMessageReceived failing. For an incoming call, the patch is first sent to Mobius then roapMessageReceived is called, in the scenario where roapMessageReceived failed since the patch is already sent the call state changes to connected even though we do not have a localAudioTrack.

## by making the following changes

- While answering, if localAudioTrack is empty call is not connected and disconnect is sent to mobius. A warning message is also given.
- While dialing, If localAudioTrack is empty, call is cleared immediately and warning message is given
- First call roapMessageReceived and then send patch to Mobius.

[dial-flow.log](https://github.com/webex/webex-js-sdk/files/13736845/dial-flow.log)
[answer-flow.log](https://github.com/webex/webex-js-sdk/files/13736846/answer-flow.log)

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested
- I removed the await in the samples app and called the answer API
- Also harcoded an empty localAudioTrack in samples and called answer API
- UTs

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/calling/src/CallingClient/calling/call.test.ts [hunk 1]
```diff
@@ -363,6 +363,89 @@ describe('Call Tests', () => {
     call.answer(localAudioStream);
     expect(call['callStateMachine'].state.value).toBe('S_SEND_CALL_CONNECT');
   });
+  it('answer fails if localAudioTrack is empty', async () => {
```
Comment (sreenara) [inline packages/calling/src/CallingClient/calling/call.test.ts:366]: please leave a blank line between tests

File: packages/calling/src/CallingClient/calling/call.ts [hunk 2]
```diff
@@ -1168,7 +1171,7 @@ export class Call extends Eventing<CallEventTypes> implements ICall {
         method: this.handleOutgoingCallAlerting.name,
       });
     } catch (err) {
-      log.warn('Call Progress failed with Mobius', {
+      log.warn('Failed to progress the call', {
```
Comment (sreenara) [inline packages/calling/src/CallingClient/calling/call.ts:1174]: This can be reworded to "Failed to signal call progression"

**Review by sreenara — CHANGES_REQUESTED** [review]
Minor comments with text

File: packages/calling/src/CallingClient/calling/call.test.ts (hunk 1 above)
Comment (Shreyas281299) [reply packages/calling/src/CallingClient/calling/call.test.ts:367]: Added.

File: packages/calling/src/CallingClient/calling/call.ts (hunk 2 above)
Comment (Shreyas281299) [reply packages/calling/src/CallingClient/calling/call.ts]: Changed.

File: packages/calling/src/CallingClient/calling/call.ts [hunk 3]
```diff
@@ -1996,6 +2015,17 @@ export class Call extends Eventing<CallEventTypes> implements ICall {
...
   public async dial(localAudioStream: LocalMicrophoneStream) {
     const localAudioTrack = localAudioStream.outputStream.getAudioTracks()[0];
+    if (!localAudioTrack) {
+      log.warn(`Did not find a local track while dialling the call ${this.getCorrelationId()}`, {
+        file: CALL_FILE,
+        method: 'dial',
+      });
+
+      this.deleteCb(this.getCorrelationId());
+      this.emit(CALL_EVENT_KEYS.DISCONNECT, this.getCorrelationId());
```
Comment (sreenara) [inline packages/calling/src/CallingClient/calling/call.ts:2025]: Nice one @Kesari3008 @Shreyas281299

**Review by sreenara — APPROVED** [review]

**Pull request merged**